### PR TITLE
Add gangplank helper

### PR DESCRIPTION
**Pull request by ravanelli**
- Add a helper to facilitate gangplank calls

Signed-off-by: Renata Ravanelli <rravanel@redhat.com>

### CODE REVIEW

File: vars/gangplank.groovy [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// Run gangplank
+def ganplankCmd
```
Comment (jlebon) [inline vars/gangplank.groovy:2]: Hmm, seems like we could pretty easily get rid of this global var by having `getMode()` return it and then passing it in to the various helpers and then finally `runGangplank()`.

File: vars/gangplank.groovy [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+def buildArtifact(params = [:]) {
+// Available parameters:
```
Comment (jlebon) [inline vars/gangplank.groovy:5]: For consistency with the other groovy commands, let's put the docstring above the functions?

File: vars/gangplank.groovy [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//     artifacts:    list    -- list of artifacts to be built
+//     extraFlags    list    -- Extra flags to use
+//     image:        string  -- Name of the image to be used
+//     workDir:      string  -- cosa working directory
```
Comment (jlebon) [inline vars/gangplank.groovy:9]: Missing `mode`

File: vars/gangplank.groovy [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        currentBuild.result = 'ABORTED'
+        error("Error running gangplank.")
+    }
+}
+
+def runSpec(params =[:]) {
+// Available parameters:
+//     artifacts:    list    -- list of artifacts to be built
+//     extraFlags    list    -- Extra flags to use
+//     spec          string  -- Name of the spec file
```
Comment (jlebon) [inline vars/gangplank.groovy:48]: Missing `mode` here too.

**Review by darkmuggle — COMMENTED** [review]
For the RHCOS and Developer use-cases we'll need to support `--git-{ref,url}` and `--repo` as well -- and then we can likely start a parallel RHCOS pipeline using the FCOS pipeline.

**Comment by ravanelli** [comment]
> For the RHCOS and Developer use-cases we'll need to support `--git-{ref,url}` and `--repo` as well -- and then we can likely start a parallel RHCOS pipeline using the FCOS pipeline.

For this case, there is the extraFlags you can pass whatever you want. But let me know if you think it would be better adding it explicitly.

**Comment by darkmuggle** [comment]
This is good and I think we can merge as is but we'll need one more command to run _after_ all the gangplank have come back: `cosa meta --finalize` or `gangplank pod -A finalize`  to bring all the `meta.*.json` files into a single `meta.json`.

**Comment by darkmuggle** [comment]
/lgtm

Great work @ravanelli!

File: vars/gangplank.groovy [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    runGangplank(gangplankCmd)
+    return fileName
+}
+
+def startParallel(configFile, gangplankCmd) {
+    gangplankCmd += "-m ${configFile}"
+    runGangplank(gangplankCmd)
+}
+
+def startMinio(params =[:]) {
```
Comment (jlebon) [inline vars/gangplank.groovy:62]: I think basically every function which takes the generic `params` arg and is intended to be "user-facing" should have a docstring with supported params. We can add one to this one in a follow-up! :)

File: vars/gangplank.groovy [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    return gangplankCmd
+}
+
+def getWorkDir(params = [:], gangplankCmd) {
+    if(params['cosaDir']) {
+        gangplankCmd += "--workDir ${utils.getCosaDir(params)} "
+        return gangplankCmd
+    }
+    return gangplankCmd
+}
```
Comment (jlebon) [inline vars/gangplank.groovy:142]: If these are more implementation details, I wonder if we should prefix them with e.g. `_` instead. WDYT? Not a blocker.

**Pull request merged**